Fix to Required/Conditionally Required/Optional field coding

antrim opened the pull request:
The following fields have been recoded as Conditionally Required:
agency.txt: `agency_id` 
stops.txt: `zone_id`
routes.txt: `agency_id`
fare_attributes.txt: `agency_id`
stop_times.txt: `arrival_time` and `departure_time`

This is open for discussion before voting.

antrim commented:
@giocorti's changes should be covered under an organizational CLA for Trillium Solutions, Inc. (@trilliumtransit).

mgilligan commented on gtfs/spec/en/reference.md:
```diff
@@ -328,7 +328,7 @@ File: **Optional**
 |   |  | | * **2** - Passenger may transfer twice. |
 |   |  | | * **(empty)** - If this field is empty, unlimited transfers are permitted. |
-|  agency_id | ID | Optional | Required for feeds with multiple agencies defined in the agency.txt file. Each fare attribute must specify an agency_id value to indicate which agency the fare applies to. |
+|  agency_id | ID | **Conditionally required** | Required for feeds with multiple agencies defined in the agency.txt file. Each fare attribute must specify an agency_id value to indicate which agency the fare applies to. This field is required for feeds that contain more than one transit agency in [agency.txt](#agencytxt), otherwise it is optional.|
```
I think this should remain `Optional`. If all agencies included in the feed have the same fare attributes & rules, they would need to duplicate everything to meet the conditional requirement.

abyrd replied:
@mgilligan my reading of the existing spec is that this field is already required for feeds with multiple agencies. I don't think the proposed changes modify the meaning of the specification, they just make it more clear.

abyrd replied:
The new language is repetitive though - it says almost exactly the same thing twice.

mgilligan replied:
@abyrd, whoops, I didn't read the original in the diff.

@antrim, please ignore my comment. :)

slai commented on gtfs/spec/en/reference.md:
```diff
@@ -219,14 +219,14 @@ File: **Required**
 |  ------ | ------ | ------ | ------ |
 |  trip_id | ID | **Required** | The **trip_id** field contains an ID that identifies a trip. This value is referenced from the [trips.txt](#tripstxt) file. | 
-|  arrival_time | Time | **Required** | The **arrival_time** specifies the arrival time at a specific stop for a specific trip on a route. The time is measured from "noon minus 12h" (effectively midnight, except for days on which daylight savings time changes occur) at the beginning of the service day. For times occurring after midnight on the service day, enter the time as a value greater than 2 ...
+|  arrival_time | Time | **Conditionally required** | The **arrival_time** specifies the arrival time at a specific stop for a specific trip on a route. The time is measured from "noon minus 12h" (effectively midnight, except for days on which daylight savings time changes occur) at the beginning of the service day. For times occurring after midnight on the service day, enter the time as a value  ...
```
This is somewhat related - I'd consider moving everything that describes when this field is required into a separate paragraph for clarity and ease of reading. At the moment, that's spread over two paragraphs, interspersed with other things.

Maybe something like this -

```
<br/><br/>This field is required for all stops that are time points and the first and last stop in a trip, otherwise it is optional. If you don't have separate times for arrival and departure at a stop, enter the same value for **arrival_time** and **departure_time**.<br/></br>
```

It may also be worth emphasising the requirement that `arrival_time` must be specified if `departure_time` is specified and vice-versa.

giocorti replied:
I agree that the arrival_time description is not very readable and should be amend. However, that's something @antrim and I hope to address in a separate PR that will also make a number of other editorial changes.

antrim commented:
All, @giocorti committed changes that addressed the comments on this pull request.

I believe the required 7 days of discussion has elapsed. Let's start the 7 day voting period for this change, unless anyone has other comments. So that there are 7 full calendar days, then the vote would end on Friday, November 9 @ 23:59:59 UTC.

slai commented:
+1

barbeau commented:
+1

abyrd commented:
+1 looks good

gcamp commented:
+!

prhod commented:
+1

antrim commented:
We have 3 votes in favor so this passes.

The CLA check is currently failing so this cannot be merged. @giocorti's contributions are covered under an organization CLA for Trillium Solutions, Inc. I have asked Google to establish this in their system.
